src/main/java/com/google/sps/servlets/FindLocationServlet.java

```diff
@@ -19,7 +27,25 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    //TODO - implement the rest of this function (@Kaleab)
+    String correlator = request.getParameter("correlator");
+
+    //TODO: find a more graceful way to handle this
+    if (correlator == null){
+      throw new IOException();
+    }
+    
+    Filter correlatorFilter =  new FilterPredicate("correlator", FilterOperator.EQUAL, correlator);
+    Query impressionQuery = new Query("Impressions").setFilter(correlatorFilter);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery filteredImpression = datastore.prepare(impressionQuery);
```

Comment:
In the advertising world impression means something very specific (a user has viewed an ad).  Correct me if I'm wrong but is this service only called on a click? If so use the word click instead of impression here and elsewhere.